metricflow/model/objects/data_source.py

```diff
@@ -64,59 +63,37 @@ class DataSource(HashableBaseModel, ParseableObject):
     origin: DataSourceOrigin = DataSourceOrigin.SOURCE
 
     @property
-    def elements(self) -> List[Element]:  # noqa: D
-        res: List[Element] = []
-        res.extend(self.identifiers)
-        res.extend(self.dimensions)
-
-        return res
-
-    @property
-    def element_names(self) -> List[LinkableElementReference]:  # noqa: D
-        return [n.name for n in self.elements]
+    def identifier_references(self) -> List[LinkableElementReference]:  # noqa: D
+        return [i.reference for i in self.identifiers]
 
     @property
-    def identifier_names(self) -> List[LinkableElementReference]:  # noqa: D
-        return [i.name for i in self.identifiers]
+    def dimension_references(self) -> List[LinkableElementReference]:  # noqa: D
+        return [i.reference for i in self.dimensions]
 
     @property
-    def dimension_names(self) -> List[LinkableElementReference]:  # noqa: D
-        return [i.name for i in self.dimensions]
-
-    @property
-    def measure_names(self) -> List[MeasureReference]:  # noqa: D
-        return [i.name for i in self.measures]
+    def measure_references(self) -> List[MeasureReference]:  # noqa: D
+        return [i.reference for i in self.measures]
 
     def get_measure(self, measure_name: MeasureReference) -> Measure:  # noqa: D
```

Comment:
I think it'd be clearer if we changed the parameter name from `measure_name` to `measure_reference`

Reply:
Updated.